cmd/controller-manager/app/controllermanager.go

```diff
@@ -591,6 +595,36 @@ func startCronFederatedHorizontalPodAutoscalerController(ctx controllerscontext.
 	return true, nil
 }
 
+func startHPAReplicasSyncerController(ctx controllerscontext.Context) (enabled bool, err error) {
+	scaleKindResolver := scale.NewDiscoveryScaleKindResolver(ctx.KubeClientSet.Discovery())
+	httpClient, err := rest.HTTPClientFor(ctx.Mgr.GetConfig())
+	if err != nil {
+		return false, err
+	}
+
+	mapper, err := apiutil.NewDiscoveryRESTMapper(ctx.Mgr.GetConfig(), httpClient)
+	if err != nil {
+		return false, err
+	}
+
+	scaleClient, err := scale.NewForConfig(ctx.Mgr.GetConfig(), mapper, dynamic.LegacyAPIPathResolverFunc, scaleKindResolver)
+	if err != nil {
+		return false, err
+	}
+
+	hpaReplicasSyncer := hpareplicassyncer.HPAReplicasSyncer{
+		Client:      ctx.Mgr.GetClient(),
+		RESTMapper:  mapper,
```

Comment:
Can we use `ctx.Mgr.GetRESTMapper()` to replace the `mapper`?

Reply:
Done.